#164523178 User Should be able edit question they posted

Philipotieno opened the pull request:
#### What does this PR do?
- Has view function to edit questions posted on meetups
- Has test for the method tto edit questions
#### Description of Task to be completed?
- `PUT api/meetups/{meetup_id}/questions/{question_id}`
#### How should this be manually tested?
- `$ git clone https://github.com/Ogutu-Brian/questioners.git`
- `$ git fetch origin` - To get all branches
- `$ git checkout ft-edit-question-164523178`
- Create a virtual environment and activate it
        `$ virtualenv -p python3 venv`
        ` $ source venv/bin/activate`
- Add a .env file as in the .env.example
        `$ source .env`
- ` $ pip intsall -r requirements.txt`
- Start server ` $ python api/manage.py runserver`
- Test the following endpoints on postman
   - Create superuser account `$ api/manage.py createsuperuser`
   - Sign in `/api/auth/signin/`
   - Create a meetup `/api/meetups/`
   - Post a question `/api/meetups/meetup_id/question`
   - Edit a question `/api/question/question_id`
- Run tests ` $ python api/manage.py test`
#### What are the relevant pivotal tracker stories?
[#164523178](https://www.pivotaltracker.com/n/projects/2315729/stories/164523178)
#### Questions:
 - Only users who own the question are allowed to edit, Would you who is reading this guide me on how to write tests for a user who is registered and logged in but not authorised to edit a question.
#### Screenshots
- Invalid question_id
![wrongid](https://user-images.githubusercontent.com/29012135/54376969-44366e00-4695-11e9-8199-ea3fb334497f.png)

- Valid Question_id
![correct](https://user-images.githubusercontent.com/29012135/54377038-69c37780-4695-11e9-801c-19e01e9730ce.png)

Amoswachira reviewed (approved):
Looks good

Ogutu-Brian reviewed (approved):
Looks good to me

a-braham reviewed (commented):
Nice job man

Philipotieno commented:
Thanks @Ogutu-Brian @Amoswachira @a-braham

Bakley commented on api/questions/tests/basetests.py:
```diff
@@ -41,7 +40,11 @@ def setUp(self):
             location='Andela Campus',
             creator=self.user,
             scheduled_date=timezone.now() + timezone.timedelta(days=3)
+<<<<<<< HEAD
```
@Philipotieno Fix this please

Philipotieno replied:
Fixed it

IssaIan reviewed (approved):
Good work.

Philipotieno committed:
feat(edit question): user should be able to edit a question

- update readme
- validate title and body
- add response messages and status codes
- Fix merge conflicts
- Merge conflicts
- Pep8 refactor
- Fix URL patterns
- Fix failing tests
- Adds swagger documentation
- Fix merge conflicts
- validate questions
[Delivers #164523178]Fix merge conflits

Ogutu-Brian commented on api/questions/views.py:
```diff
@@ -70,3 +70,45 @@ def get(self, request, id=None):
+                        body=request.data.get('body')
+                    )
+                    return Response({"message":"question updated succesfully"}, status=status.HTTP_201_CREATED)
+                return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```
Hello @a-braham ,
This is a great work. I was wondering whether you could the number of return statements per method or function to one. It would make your methods and functions have a single entry point and a single exit point.

davenjeru commented on README.md:
````diff
@@ -56,7 +56,11 @@
  ```
-
+## Questions Endpoints
+- User edit questio
````
Poor attention to detail here. Please correct the typo here.

Philipotieno replied:
Correcting it

davenjeru commented on api/config/swagger.py:
```diff
@@ -378,7 +378,7 @@ def schema_view_swagger(request):
                 'Post question': coreapi.Link(
                     # fetch meetup id from the database
-                    url='/api/meetup/{meetup_id}/question/',
+                    url='/api/meetups/{meetup_id}/question/',
```
Use plural on the endpoint as discussed on the slack channel.

Philipotieno replied:
working on this too

Philipotieno commented:
@davenjeru All checks have passed
